src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -1706,12 +1706,22 @@ public Uri MakeRelativeUri(Uri uri)
         // 
         private static bool CheckForColonInFirstPathSegment(string uriString)
         {
-            // Check for anything that may terminate the first regular path segment
-            // or an illegal colon
-            char[] pathDelims = new char[] { ':', '\\', '/', '?', '#' };
-            int index = uriString.IndexOfAny(pathDelims);
-
-            return (index >= 0 && uriString[index] == ':');
+            for (int i = 0; i < uriString.Length; i++)
```

Comment:
This is saying that we're having a bug in `string.IndexOfAny`. If that's true, open an issue and, ideally, add a disabled `string` perf test to prove the problem.